conjure-http/src/client.rs

```diff
@@ -328,7 +328,39 @@ pub trait AsyncDeserializeResponse<T, R> {
     fn accept() -> Option<HeaderValue>;
 
     /// Deserializes the response.
-    async fn deserialize(response: Response<R>) -> Result<T, Error>;
+    async fn deserialize(response: Response<R>) -> Result<T, Error>
+    where
+        R: 'async_trait;
```

Comment:
This isn't a breaking change, since implementations that don't need to support the shorter lifetime can omit the where clause.